tools/crd-bumper/pkg/git_cli.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2024 Hewlett Packard Enterprise Development LP
+# Other additional copyright holders may be indicated within.
+#
+# The entirety of this work is licensed under the Apache License,
+# Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License.
+#
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import subprocess
+
+from .copyright import Copyright
+
+class GitCLI:
+
+  def __init__(self, dryrun, nocommit):
+    self._dryrun = dryrun
+    if dryrun:
+        nocommit = True
+    self._nocommit = nocommit
+
+  def verify_branch(self, branch):
+    cmd = "git status | sed 1q | awk '{{print $3}}'"
+    child = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
```

Comment:
Why not use subprocess.check_output() for these commands?

Reply:
check_output returns stdout only. Sometimes I want to collect the stderr.

Reply:
You should be able to with: `check_output(stderr=subprocess.STDOUT)`. Or in the error case, you can pull stderr from the `CallProcessError` exception.